src/cloudai/report_generator/tool/bokeh_report_tool.py

```diff
@@ -195,13 +201,26 @@ def add_log_x_linear_y_single_line_plot(
         x_min, x_max = self.find_min_max(df, x_column)
         y_min, y_max = self.find_min_max(df, y_column, sol)
 
+        # Check if x_min equals x_max - constant message size
+        if x_min == x_max:
+            # Use iteration number as x-axis
+            df["iteration"] = range(1, len(df) + 1)
+            x_column = "iteration"
+            x_axis_label = "Iteration"
+            x_axis_type = "linear"
+            x_range = Range1d(start=1, end=len(df))
+        else:
```

Comment:
_[minor]_ instead of if-else it could be default + if:
```py
x_axis_type = "log"
x_range = None
if x_min == x_max:  # constant message size
    ...
```